cli/file_fetcher.rs

```diff
@@ -213,54 +211,38 @@ impl SourceFileFetcher {
   ///
   /// If `cached_only` is true then this method will fail for remote files
   /// not already cached.
-  fn get_source_file_async(
+  async fn get_source_file_async(
     &self,
     module_url: &Url,
     use_disk_cache: bool,
     no_remote: bool,
     cached_only: bool,
-  ) -> impl Future<Output = Result<SourceFile, ErrBox>> {
+  ) -> Result<SourceFile, ErrBox> {
     let url_scheme = module_url.scheme();
     let is_local_file = url_scheme == "file";
-
-    if let Err(err) = SourceFileFetcher::check_if_supported_scheme(&module_url)
-    {
-      return Either::Left(futures::future::err(err));
-    }
+    SourceFileFetcher::check_if_supported_scheme(&module_url)?;
 
     // Local files are always fetched from disk bypassing cache entirely.
     if is_local_file {
-      match self.fetch_local_file(&module_url) {
-        Ok(source_file) => {
-          return Either::Left(futures::future::ok(source_file));
-        }
-        Err(err) => {
-          return Either::Left(futures::future::err(err));
-        }
-      }
+      return self.fetch_local_file(&module_url);
     }
 
     // The file is remote, fail if `no_remote` is true.
     if no_remote {
-      return Either::Left(futures::future::err(
-        std::io::Error::new(
-          std::io::ErrorKind::NotFound,
-          format!(
-            "Not allowed to get remote file '{}'",
-            module_url.to_string()
-          ),
-        )
-        .into(),
-      ));
+      let e = std::io::Error::new(
+        std::io::ErrorKind::NotFound,
+        format!(
+          "Not allowed to get remote file '{}'",
+          module_url.to_string()
+        ),
+      );
```

Comment:
Aside, not for this patch: basically anywhere we're constructing errors we should be using some predefined utility functions from cli/deno_error.rs. So this code would ideally be written:
```
deno_error::not_found(format!("Not allowed to get remote file '{}'", module_url.to_string())
```

Reply:
Yes!!! Completely agree, this becomes very obvious working on #3936